iOS_SDK/OneSignalSDK/Source/OSStateSynchronizer.m

```diff
@@ -148,6 +148,13 @@ - (void)registerUserWithState:(OSUserState *)registrationState withSuccess:(OSMu
         //update push player id
         if (results.count > 0 && results[OS_PUSH][@"id"]) {
             self.currentSubscriptionState.userId = results[OS_PUSH][@"id"];
+            
+            // Player record was deleted so we should reset external user id
+            let cachedPlayerId = [OneSignalUserDefaults.initStandard getSavedStringForKey:OSUD_PLAYER_ID_TO defaultValue:nil];
```

Comment:
When the player is deleted from the server on Android, we clean userId and all cached data for that player. Finally, re-trigger the player's creation